AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -224,47 +246,31 @@ - (void)setFrame:(CGRect)rect
     // Checking if the transform is identity is expensive, so disable when unnecessary. We have assertions on in Release, so DEBUG is the only way I know of.
     ASDisplayNodeAssert(CATransform3DIsIdentity(self.transform), @"-[ASDisplayNode setFrame:] - self.transform must be identity in order to set the frame property.  (From Apple's UIView documentation: If the transform property is not the identity transform, the value of this property is undefined and therefore should be ignored.)");
 #endif
+    _view.frame = rect;
+  } else if (!nodeLoaded || (isMainThread && !setFrameDirectly)) {
+    /**
+     * Sets a new frame to this node by changing its bounds and position. This method can be safely called even if
+     * the transform is a non-identity transform, because bounds and position can be set instead of frame.
+     * This is NOT called for synchronous nodes (wrapping regular views), which may rely on a [UIView setFrame:] call.
+     * A notable example of the latter is UITableView, which won't resize its internal container if only layer bounds are set.
+     */
+    CGRect bounds;
+    CGPoint position;
+    ASBoundsAndPositionForFrame(rect, self.bounds.origin, self.anchorPoint, &bounds, &position);
 
-    _setToViewOnly(frame, rect);
-  } else {
-    // This is by far the common case / hot path.
-    [self __setSafeFrame:rect];
-  }
-}
-
-/**
- * Sets a new frame to this node by changing its bounds and position. This method can be safely called even if
- * the transform is a non-identity transform, because bounds and position can be set instead of frame.
- * This is NOT called for synchronous nodes (wrapping regular views), which may rely on a [UIView setFrame:] call.
- * A notable example of the latter is UITableView, which won't resize its internal container if only layer bounds are set.
- */
-- (void)__setSafeFrame:(CGRect)rect
-{
-  ASDisplayNodeAssertThreadAffinity(self);
-  ASDN::MutexLocker l(_propertyLock);
-  
-  BOOL useLayer = (_layer && ASDisplayNodeThreadIsMain());
-  
-  CGPoint origin      = (useLayer ? _layer.bounds.origin : self.bounds.origin);
```

Comment:
I'm pretty sure it was a profiled optimization to use the layer accessor when possible, and we should preserve this. Going through the view simply adds more method calls, because it must go to the layer anyway.

In general, the modifications to this frame code are my biggest concern with the patch so far. It seems very plausible that they are correct, but I think it adds risk to an already large diff that is not strictly necessary. If there is a way to start with the original code and make a minimum of modification so it is clear what has changed, that would be my preference. This won't block us from landing though, just have some nightmares about this code and don't feel like Messing around with it too much right now :)